### ISSUE TITLE
KeyError: 'ussd_service_op' for ussd command

### ISSUE THREAD

**Issue by ron-15** [python-smpplib/python-smpplib#104]
I use your smpplib for python library. As I understand it, the library supports the library from the ussd source code. At the moment, from our ussdc sent to the client.listen () port. It listens correctly without failures, but when pdu goes, the listener crashes, with an error:

**File "C:\Users\AppData\Local\Programs\Python\Python37-32\lib\site-packages\smpplib\command.py", line 332, in parse_optional_params
    param = self.params[field]
KeyError: 'ussd_service_op'
WARNING:smpplib.client:<smpplib.client.Client object at 0x029DBCB0>
was not closed**

My script:
_import logging
import sys

import smpplib.gsm
import smpplib.client
import smpplib.consts

logging.basicConfig(level='DEBUG')
client = smpplib.client.Client('host', 8884, )
client.connect()
print('Коннект! ')
client.bind_transceiver(system_id='login', password='pass')

client.set_message_sent_handler(
    lambda pdu: sys.stdout.write('sent {} {}\n'.format(pdu.sequence, pdu.message_id)))
client.set_message_received_handler(
    lambda pdu: sys.stdout.write('delivered {}\n'.format(pdu.receipted_message_id)))

print(client.read_pdu())

client.listen()_

Which method should I use to hold a ussd session read Delivery_sm and Submit_sm ?

**Comment by eigenein** [comment]
Looks like we're actually missing the `ussd_service_op` field in our `SubmitSM` and `DeliverySM` definitions. This has to be fixed

**Comment by eigenein** [comment]
Alright, this is even simpler than I thought. USSD support is already in place for `SubmitSM`, but we're missing it for `DeliverySM`.

@ron-15 You can try to replicate these two lines:
- https://github.com/python-smpplib/python-smpplib/blob/master/smpplib/command.py#L677
- https://github.com/python-smpplib/python-smpplib/blob/master/smpplib/command.py#L698

for `DeliverySM` class: https://github.com/python-smpplib/python-smpplib/blob/master/smpplib/command.py#L730

In other words, copy-paste `ussd_service_op` definition to `params` dict and `params_order` list. That should be it

**Comment by ron-15** [comment]
Did what you asked above.
Also added the ussd_session_id parameter (Parameter Tag 2 Integer ussd_session_id (0x1501)) to the SubmitSM and DeliverySM classes. The previous error is gone.
But now with a ussd session (for example, with the code 0x1915, each time it is different) - it displays an error:
_smpplib.exceptions.UnknownCommandError: Unknown SMPP command code "0x1915"._
Where in pars can I separate the session id from the team itself?

**Comment by eigenein** [comment]
Looks like you implemented it correct.

But now you have a totally different issue, similar to #2. This is most likely a vendor-specific command code, which is not included in SMPP standard. Perhaps here you should talk to your SMPP provider and ask what does the command `0x1915` really mean
